ultraplot/tests/test_geographic.py

```diff
@@ -164,9 +164,181 @@ def test_geoticks_input_handling(recwarn):
+        return on
+
+    n = 3
+    settings = dict(land=True, ocean=True, labels="both")
+    fig, ax = uplt.subplots(ncols=n, nrows=n, share="all", proj="cyl")
+    ax.format(**settings)
+    fig.canvas.draw()  # need a draw to trigger ax.draw for  sharing
+
+    expectations = (
+        [True, False, False, True],
+        [True, False, False, False],
+        [True, False, True, False],
+        [False, False, False, True],
+        [False, False, False, False],
+        [False, False, True, False],
+        [False, True, False, True],
+        [False, True, False, False],
+        [False, True, True, False],
+    )
+    for axi in ax:
+        state = are_labels_on(axi)
+        expectation = expectations[axi.number - 1]
+        for i, j in zip(state, expectation):
+            assert i == j
+
+    layout = [
+        [1, 2, 0],
+        [1, 2, 5],
+        [3, 4, 5],
+        [3, 4, 0],
+    ]
+
+    fig, ax = uplt.subplots(layout, share="all", proj="cyl")
+    ax.format(**settings)
+    fig.canvas.draw()  # need a draw to trigger ax.draw for  sharing
+
+    expectations = (
+        [True, False, False, True],  # top left
+        [True, False, True, False],  # top right
+        [False, True, False, True],  # bottom left
+        [False, True, True, False],  # bottom right
+        [True, True, True, False],  # right plot (5)
+    )
+    for axi in ax:
+        state = are_labels_on(axi)
+        expectation = expectations[axi.number - 1]
+        assert all([i == j for i, j in zip(state, expectation)])
+    return fig
+
+
+def test_toggle_gridliner_labels():
+    """
+    Test whether we can toggle the labels on or off
+    """
+    # Cartopy backend
+    fig, ax = uplt.subplots(proj="cyl", backend="cartopy")
+    ax[0]._toggle_gridliner_labels(left=False, bottom=False)
+    gl = ax[0].gridlines_major
+
+    assert gl.left_labels == False
+    assert gl.right_labels == False
+    assert gl.top_labels == False
+    assert gl.bottom_labels == False
+    ax[0]._toggle_gridliner_labels(top=True)
+    assert gl.top_labels == True
+    uplt.close(fig)
+
+    # Basemap backend
+    fig, ax = uplt.subplots(proj="cyl", backend="basemap")
+    ax.format(land=True, labels=True)  # need this otherwise no labels are printed
+    ax[0]._toggle_gridliner_labels(left=False, bottom=False)
+    gl = ax[0].gridlines_major
+
+    # All label are off
+    for gli in gl:
+        for _, (line, labels) in gli.items():
+            for label in labels:
+                assert label.get_visible() == False
+
+    # Should be off
+    ax[0]._toggle_gridliner_labels(top=True)
+    # Gridliner labels are not added for the top (and I guess right for GeoAxes). Need to figure out how this is set is set in matplotlib
+    dir_labels = ax[0]._get_gridliner_labels(
+        left=True, right=True, top=True, bottom=True
+    )
+    for dir, labels in dir_labels.items():
+        expectation = False
+        if dir == "top":
+            expectation = True
```

Comment:
This line of code is missed in code coverage which means I think that this test is not quite working correctly.

Reply:
Thanks! Fixed. Apparently the labels need to be toggled first before they can be turned off. Will investigate this as it is not intended. Keeping it open for now, will fix the other stuff first.